libusb/os/windows_winusb.c

```diff
@@ -1480,6 +1480,110 @@ static int set_hid_interface(struct libusb_context *ctx, struct libusb_device *d
 	return LIBUSB_SUCCESS;
 }
 
+static int get_guid(struct libusb_context *ctx, char *dev_id, HDEVINFO *dev_info, SP_DEVINFO_DATA *dev_info_data,
```

Comment:
For future use, maybe return number of guids discovered on success?

Reply:
Do I understand correctly that dev_id is only used for the warning messages? I guess that is still worth it though.

Reply:
> For future use, maybe return number of guids discovered on success?
There is a parameter 'guid_number', which should ensure taking the n-th GUID. So for going through all GUIDs it is possible to increasing the guid_number as long as the get_guid() returns SUCCESS. From my angle, returning number of GUIDs is a little strange. However if you wish, I have no problem with it.

> Do I understand correctly that dev_id is only used for the warning messages? I guess that is still worth it though.
Yes, the dev_id is used only for the warning messages. It makes me sense to keep it here.

Reply:
You're probably right, there is no potential use of returning the number of GUIDs. Iterating over this function like you suggest, would work well.